### PR TITLE
SIP50: Web Socket API

### PR DESCRIPTION
**Pull request by ohager**
This documents proposes a new lightweight real time API based on WebSockets for the reference software.

### CODE REVIEW

File: SIP/sip-50.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+| Field             | Type      | Description                                                 | Example                  |
+|-------------------|-----------|-------------------------------------------------------------|--------------------------|
+| `e`               | String    | Represents the event type or action. Event name SHOULD be uppercase. | `"CONNECTED"`            |
+| `p`               | Object    | Payload containing relevant data for the event. This field is optional              |                          |
+
+#### Event `CONNECTED`
+
+This event MUST be emitted by the Node Software upon every established websocket connection.
+
+The payload response SHOULD have ate least the following fields:
```
Comment (ipr0310) [inline SIP/sip-50.md:52]: I think it should be `at least`

File: SIP/sip-50.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  "e": "HEARTBEAT",
+}
+
+```
+
+#### Event `BLOCK_PUSHED`
+
+This event SHALL be emitted by the Node Software upon pushed blocks,i.e. while the node is syncing. As the node can push several dozen to hundred blocks per second, the event emission MUST be debounced to a configurable duration, e.g. one second, where only the latest pushed block SHALL be submitted.
+
+The payload response SHOULD have ate least the following fields:
```
Comment (ipr0310) [inline SIP/sip-50.md:102]: I think it should be `at least`

File: SIP/sip-50.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    "transactionCount": 5
+  }
+}
+```
+
+#### Event `PENDING_TRANSACTIONS_ADDED`
+
+This event SHALL be emitted by the Node Software upon pushed blocks,i.e. while the node is syncing. As the node can push several dozen to hundred blocks per second, the event emission MUST be debounced to a configurable duration, e.g. one second, where only the latest pushed block SHALL be submitted.
+
+The payload response SHOULD have ate least the following fields:
```
Comment (ipr0310) [inline SIP/sip-50.md:135]: I think it should be `at least`

**Review by ipr0310 — COMMENTED** [review]
Really interesting proposal.
Just found small grammar errors.

**Comment by frankTheTank72** [comment]
Please add also to the main page from the SIPs @ohager

File: SIP/sip-50.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+author: ohager
+status: Draft
+type: standard
+category: Interface, SRC
+created: 2024-01-10
+---
+
+
+## Abstract
+This document describes the communication interface for lighweight realtime communication via WebSockets for the Node Software.
```
Comment (frankTheTank72) [inline SIP/sip-50.md:14]: This document describes the lightweight real-time communication interface via WebSockets for the Node Software.

File: SIP/sip-50.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+The responses from the Node Software follow a unified structure. All responses MUST be serialized valid JSON (RFC8259)
+
+| Field             | Type      | Description                                                 | Example                  |
+|-------------------|-----------|-------------------------------------------------------------|--------------------------|
+| `e`               | String    | Represents the event type or action. Event name SHOULD be uppercase. | `"CONNECTED"`            |
+| `p`               | Object    | Payload containing relevant data for the event. This field is optional              |                          |
+
+#### Event `CONNECTED`
+
+This event MUST be emitted by the Node Software upon every established websocket connection.
```
Comment (frankTheTank72) [inline SIP/sip-50.md:50]: The Node Software MUST emit this event upon every established websocket connection.

**Comment by ohager** [comment]
> Please add also to the main page from the SIPs @ohager

done

**Review by frankTheTank72 — APPROVED** [review]

**Pull request merged**